SwanFileBrowser/src/kernelspec/validate.ts

```diff
@@ -0,0 +1,198 @@
+/**
+ * File took from https://github.com/jupyterlab/jupyterlab/blob/master/packages/services/src/kernelspec/validate.ts
+ * for jupyterlab 3.0.x
+ *
+ * This allows to validate the kernels specs and to get the ISpecModel.
+ * This is required to update the manager.services.kernelspecs.specs object
+ * in the SwanFileBrowserModel.
```

Comment:
Would it make sense to mention this file in the index.ts summary, too? There you mention listing.ts and swanfilebrowser.ts.

Reply:
Yes, agree, I did it.